pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1303,6 +1312,10 @@ func buildDefaultCluster(push *model.PushContext, name string, discoveryType api
 	}
 
 	if discoveryType == apiv2.Cluster_STATIC || discoveryType == apiv2.Cluster_STRICT_DNS {
+		if direction == model.TrafficDirectionOutbound && len(localityLbEndpoints) == 0 {
+			log.Errorf("STRICT_DNS/STATIC Cluster:%s, does not have endpoints, skipped", name)
+			return nil
+		}
```

Comment:
Can you explain how can this happen?

Reply:
For example:
1. configure a DNS type ServiceEntry
```yaml
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: static-20191230
  namespace: gateway-system
spec:
  endpoints:
  - address: 10.108.251.133
    ports:
      http: 80
  - address: 10.96.144.80
    ports:
      http: 80
  exportTo:
  - '*'
  hosts:
  - com.netease.static-20191230
  location: MESH_EXTERNAL
  ports:
  - name: http
    number: 80
    protocol: HTTP
  resolution: DNS
```
2. configure a subset, which cannot select any endpoint
```go
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: static-1230
  namespace: gateway-system
spec:
  host: com.netease.static-20191230
  subsets:
  - altStatName: healthCheckTest
    name: static-20191230-test1
    labels:
       app: none
```

Reply:
IC, seems you should also provide the VirtualService